packages/web/src/components/transaction-details-modal/components/TransactionDetailsContent.tsx

```diff
@@ -97,6 +97,10 @@ const UserDetails = ({ userId }: UserDetailsProps) => {
 const dateAndMetadataBlocks = (transactionDetails: TransactionDetails) => {
   switch (transactionDetails.transactionType) {
     case TransactionType.PURCHASE: {
+      // Rare case: metadata will be null if failed to fetch from identity.
+      if (transactionDetails.metadata === null) {
+        return
+      }
       return (
         <>
           <TransactionPurchaseMetadata
```

Comment:
Should be able to take out the `!` on the line below now

Reply:
lint was complaining cuz i think it could technically still be undefined.

Reply:
above check as `if (!transactionDetails.metadata)` might do it then